opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -1682,15 +1682,20 @@ const buildLocalMustFilter = async (validFilter) => {
   const valuesFiltering = [];
   const noValuesFiltering = [];
   const { key, values, nested, operator = 'eq', mode: localFilterMode = 'or' } = validFilter;
+  if (isEmptyField(key)) {
+    throw FunctionalError('A filter key must be defined', { key });
+  }
   const arrayKeys = Array.isArray(key) ? key : [key];
+  const headKey = R.head(arrayKeys);
+  const dontHandleMultipleKeys = nested || operator === 'nil' || operator === 'not_nil';
+  if (dontHandleMultipleKeys && arrayKeys.length > 1) {
+    throw UnsupportedError('Filter must have only one field', { keys: arrayKeys });
```

Comment:
```suggestion
    throw UnsupportedError('Filter must have only one field', { keys: arrayKeys, operator });
```
This way we know if its because of the operator or nested